opendata_module/opmon_opendata/api/helpers.py

```diff
@@ -0,0 +1,213 @@
+
+    data_cursor, column_names, date_columns = _get_content_cursor(postgres, date, columns, constraints, order_clauses)
+
+    gzip_buffer = BytesIO()
+    with GzipFile(fileobj=gzip_buffer, mode='wb') as gzip_file:
+        count = 0
+        buffer_size = settings['opendata'].get('stream-buffer-lines', DEFAULT_STREAM_BUFFER_LINES)
+        for row in data_cursor:
+            json_obj = {column_name: row[column_idx] for column_idx, column_name in enumerate(column_names)}
+            # Must manually convert Postgres dates to string to be compatible with JSON format
+            for date_column in date_columns:
+                json_obj[date_column] = datetime.strftime(json_obj[date_column], '%Y-%m-%d')
+            gzip_file.write(bytes(json.dumps(json_obj), 'utf-8'))
+            gzip_file.write(b'\n')
+            count += 1
+            if count == buffer_size:
+                count = 0
+                data = gzip_buffer.getvalue()
+                data_len = len(data)
+                if data_len:
+                    # Yield current buffer
+                    yield data
+                    # Empty buffer to free memory
+                    gzip_buffer.truncate(0)
+                    gzip_buffer.seek(0)
+    # Final data gets written when GzipFile is closed
+    yield gzip_buffer.getvalue()
+
+
+def get_content(postgres: PostgreSQL_Manager, date: datetime, columns: List[str],
+                constraints: List[dict], order_clauses: List[dict],
+                limit: Optional[int] = None) -> Tuple[List[Tuple], List[str], List[str]]:
+    data_cursor, columns, date_columns = _get_content_cursor(
+        postgres, date, columns, constraints, order_clauses, limit=limit)
+    return data_cursor.fetchall(), columns, date_columns
+
+
+def _get_content_cursor(postgres: PostgreSQL_Manager, date: datetime, columns: List[str],
+                        constraints: List[dict], order_clauses: List[dict],
+                        limit: Optional[int] = None) -> Tuple[psycopg2.extensions.cursor, List[str], List[str]]:
+    constraints.append({'column': 'requestInDate', 'operator': '=', 'value': date.strftime('%Y-%m-%d')})
+
+    column_names_and_types = postgres.get_column_names_and_types()
+
+    if not columns:  # If no columns are specified, all must be returned
+        columns = [column_name for column_name, _ in column_names_and_types]
+
+    date_columns = [column_name for column_name, column_type in column_names_and_types
+                    if column_type == 'date' and column_name in columns]
+    data_cursor = postgres.get_data_cursor(
+        constraints=constraints, columns=columns, order_by=order_clauses, limit=limit)
+
+    return data_cursor, columns, date_columns
+
+
+def generate_meta_file(postgres: PostgreSQL_Manager, columns: List[str],
+                       constraints: List[dict], order_clauses: List[dict], field_descriptions: dict) -> bytes:
```

Comment:
```suggestion
                       constraints: list[dict], order_clauses: list[dict], field_descriptions: dict) -> bytes:
```
Right?

Reply:
@justinask7 Project uses python3.8. at this moment. Changes you suggest were introduced in python3.9

Reply:
But I see `dict` used here then 🤔

Reply:
Yes, this is correct: using generics as types is supported in python3.8. but you can not use parameters for generics as this feature was introduced with PEP-585. You need to use `typing.List` for that.